src/frontends/onnx/tests/onnx_reader_external_data.cpp

```diff
@@ -114,3 +118,183 @@ TEST(ONNX_Reader_Tests, ImportModelWithExternalDataFromWstringNamedFile) {
     ASSERT_TRUE(external_data_node_const->get_vector<float>() == (std::vector<float>{1, 2, 3, 4}));
 }
 #endif
+
+// API 2.0 tests
+
+class OnnxFeMmapFixture : public ::testing::TestWithParam<bool> {};
+
+TEST_P(OnnxFeMmapFixture, onnx_external_data) {
+    const auto path =
+        CommonTestUtils::getModelFromTestModelZoo(std::string(ONNX_TEST_MODELS) + "external_data/external_data.onnx");
+    ov::Core core;
+    core.set_property(ov::enable_mmap(GetParam()));
+    const auto model = core.read_model(path);
```

Comment:
will we really have mmap enabled? I am curious if this logic works:

```
    const bool enable_mmap =
        variants[variants.size() - 1].is<bool>() ? variants[variants.size() - 1].as<bool>() : false;
```

Reply:
It's a equivalent of IR frontend - https://github.com/openvinotoolkit/openvino/blob/008d9a83cecbd6414587158b298c439ea9fd1e54/src/frontends/ir/src/frontend.cpp#L186, but the question if enable_mmap should be true by default is resonable. In the current state yes, because of https://github.com/openvinotoolkit/openvino/blob/008d9a83cecbd6414587158b298c439ea9fd1e54/src/inference/src/dev/core_impl.hpp#L97